.github/workflows/unit-tests.yaml

```diff
@@ -5,8 +5,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: ["3.7", "3.8", "3.9", "3.10"]
```

Comment:
Did 3.7 cause any issues?

Reply:
I removed 3.7 after the tests failed.

Reply:
hmmm, let me check logs to see what exactly failed

Reply:
Oh, I see, this is related to deprecated functionality in github actions, not necessarily our code. It makes sense to move forward, we don't have to provide any backward compatibility here and complicate how actions are run